test/destructure_test.exs

```diff
@@ -7,7 +7,23 @@ defmodule DestructureTest do
     defstruct [:name, :email]
   end
 
+  defmodule People do
+    defstruct [:people]
+  end
+
   alias __MODULE__.Person
 
   doctest Destructure
+
+  describe "structs" do
+    test "can't match with different structs" do
+      refute d(%People{}) == %Person{}
```

Comment:
No, if we use assignment match, it will fail before `refute` evaluation takes action. Try using `assert d(%People{}) == %Person{}`, we will get the expected failure
```elixir
1) test structs can't match with different structs (DestructureTest)
     test/destructure_test.exs:17
     Assertion with == failed
     code: d(%People{}) == %Person{}
     lhs:  %DestructureTest.People{people: nil}
     rhs:  %DestructureTest.Person{email: nil, name: nil}
     stacktrace:
       test/destructure_test.exs:18: (test)
```